src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,127 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+type Props = {
+  todos: Todo[],
+  onClickSelectUser: (userId: number) => void,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onClickSelectUser,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [selectValue, setSelectValue] = useState('all');
+
+  const todosSelectArrFilter = (arr: Todo[], completedValue: string) => {
+    switch (completedValue) {
+      case 'active':
+        return arr.filter(todo => todo.completed === false);
+
+      case 'completed':
+        return arr.filter(todo => todo.completed === true);
+
+      default:
+        return arr;
+    }
+  };
+
+  const todosInputFilter = (arr: Todo[], titleValue: string) => {
+    if (!titleValue) {
+      return arr;
+    }
+
+    return arr.filter(todo => todo.title.includes(titleValue));
+  };
+
+  const preparedTodosArr = () => {
+    const copyTodos = [...todos];
+
+    const filterByInput = todosInputFilter(copyTodos, inputValue);
+
+    const filterBySelect = todosSelectArrFilter(filterByInput, selectValue);
```

Comment:
no need for these separate functions, just add switch case here, then filter by title and then just return a result